File: scripts/convert_counts_to_cpm.R
# Convert to CPM (counts per million)
cpm_func <- function(counts) {
    cpm <- apply(counts,2, function(x) (x/sum(x))*1000000)
    cpm = as.matrix(cpm)
    rownames(cpm) = rownames(counts)
    colnames(cpm) = colnames(counts)
    return(cpm)
}
counts = read.table("../counts/counts_matrix.tsv",sep="\t",header=TRUE, row.names=1)
cpm = cpm_func(counts[rowSums(counts)>0,])

# Write to file
write.table(cpm, file="../counts/cpm_matrix.tsv", sep="\t", quote=FALSE, row.names = TRUE)
                 
# Function to filter CPM > 2
cpm_greaterthan2_func <- function(cpm_0) {
    check_max_cpm = apply(cpm_0, 1, max)
    cpm_2 = cpm_0[which(check_max_cpm > 2), ]
    return(cpm_2)
}

# Apply CPM > 2 transformation
cpm_2 = cpm_greaterthan2_func(cpm)
print(colSums(cpm_2>0)) # count # miRNAs detected > 2 CPM

# Write CPM > 2 matrix to file
write.table(cpm_2, file="../counts/cpm_over2_matrix.tsv", sep="\t", quote=FALSE, row.names=TRUE)

File: scripts/convert_counts_to_cpm.py
import pandas as pd

def cpm_func(counts):
    cpm = counts.apply(lambda x: (x / x.sum()) * 1e6, axis=0)
    return cpm

def cpm_greaterthan2_func(cpm_0):
    check_max_cpm = cpm_0.apply(max, axis=1)
    cpm_2 = cpm_0[check_max_cpm > 2]
    return cpm_2

# Read counts matrix
counts = pd.read_csv("../counts/counts_matrix.tsv", sep="\t", header=0, index_col=0)

# Apply CPM transformation
cpm = cpm_func(counts.loc[counts.sum(axis=1) > 0])

# Write CPM matrix to file
cpm.to_csv("../counts/cpm_matrix.tsv", sep="\t", index=True)

# Apply CPM > 2 transformation
cpm_2 = cpm_greaterthan2_func(cpm)

print((cpm_2 > 0).sum(axis=0))  # count # miRNAs detected > 2 CPM

# Write CPM > 2 matrix to file
cpm_2.to_csv("../counts/cpm_over2_matrix.tsv", sep="\t", index=True)
